travis-ci/make_rpms.sh

```diff
@@ -114,6 +114,9 @@ build_all_for_target()
             return 1
         fi
 
+        # install build requires as per https://stackoverflow.com/a/44443092
+        sudo yum install -y $( rpmbuild ${nocheck:+--nocheck} --target $target --rebuild "$rpmpath" 2>&1 | fgrep 'is needed by' | awk '{print $1}' )
```

Comment:
this looks quite ugly and would actually rebuild the $rpmpath. The output of something like `rpm --requires -qp "$rpmpath" | cut -d' ' -f1`. That's still ignoring version requirements, so still not ideal, but looks a lot cleaner.
Maybe I'm missing something...?

Reply:
I'm happy with anything better than what's in the PR. The actual problem is described in https://github.com/gridcf/UberFTP/pull/16#issuecomment-974179081. No build dependencies are installed prior to the build of the RPMs not even in [the original file](https://github.com/gridcf/gct/blob/5165cc9685d5dd5a40df445ca671f06771da8099/travis-ci/make_rpms.sh) - maybe this was never tested and the original deploy step with Travis-CI created SRPMs only, which would work despite the missing build dependencies.

Reply:
Right, so probably
```
sudo yum install -y $(rpm --requires -qp "$rpmpath" | cut -d' ' -f1)
```
is already better, except that it won't take into account requirements with version constraints.
Have you actually tried
```
sudo yum-builddep --releasever=... $rpmpath
```
I didn't succeed without the releasever set to e.g. 7, but with it it seems fine

 Not sure how to best do that.

Reply:
I'll give that a try.

****

_Sorry, apparently I didn't hit the "Comment" button until now._ :-/

Reply:
No longer needed with https://github.com/gridcf/UberFTP/pull/16/commits/bbaf36959599cf768db97e5cf189909e77ee01f2